src/phoenix/experimental/evals/models/base.py

```diff
@@ -30,35 +29,41 @@
 )
 
 
-def create_base_retry_decorator(
-    error_types: List[Type[BaseException]],
-    min_seconds: int,
-    max_seconds: int,
-    max_retries: int,
-) -> Callable[[Any], Any]:
-    """Create a retry decorator for a given LLM and provided list of error types."""
-
-    # TODO: Nice logging. The logging implemented is huge and overwhelming
-    _logging = before_sleep_log(logger, logging.WARNING)
-
-    def _before_sleep(retry_state: RetryCallState) -> None:
-        _logging(retry_state)
-        return None
-
-    retry_instance: retry_base = retry_if_exception_type(error_types[0])
-    for error in error_types[1:]:
-        retry_instance = retry_instance | retry_if_exception_type(error)
-    return retry(
-        reraise=True,
-        stop=stop_after_attempt(max_retries),
-        wait=wait_random_exponential(multiplier=1, min=min_seconds, max=max_seconds),
-        retry=retry_instance,
-        # before_sleep=_before_sleep,
-    )
-
-
 @dataclass
 class BaseEvalModel(ABC):
+    _verbose: bool = False
+
+    def retry(
```

Comment:
This is a rough attempt to clean up an abstractions: the `create_base_retry_decorator` function was imported from the `base` module into both of our concrete implementations. After attaching a `verbose` state to the model it also meant we needed to feed a property on the base model back into this function so I'm moving the decorator directly into model as an instance method. Please let me know if this feels unpleasant.

I also think it also makes sense if we use the factory directly as a decorator